constraint/pkg/client/drivers/k8scel/schema/schema.go

```diff
@@ -65,6 +99,56 @@ func (in *Source) GetMatchConditions() ([]cel.ExpressionAccessor, error) {
 	return matchConditions, nil
 }
 
+func (in *Source) GetV1Alpha1MatchConditions() ([]admissionv1alpha1.MatchCondition, error) {
+	var matchConditions []admissionv1alpha1.MatchCondition
+	for _, mc := range in.MatchConditions {
+		matchConditions = append(matchConditions, admissionv1alpha1.MatchCondition{
+			Name:       mc.Name,
+			Expression: mc.Expression,
+		})
+	}
+	return matchConditions, nil
+}
+
+func (in *Source) validateVariables() error {
+	for _, v := range in.Variables {
+		if strings.HasPrefix(v.Name, ReservedPrefix) {
+			return fmt.Errorf("%w: %s is not a valid variable; cannot have %q as a prefix", ErrBadVariable, v.Name, ReservedPrefix)
+		}
+		if v.Name == ParamsName {
+			return fmt.Errorf("%w: %s an invalid variable name, %q is a reserved keyword", ErrBadVariable, ParamsName, ParamsName)
+		}
+	}
+	return nil
+}
+
+func (in *Source) GetVariables() ([]cel.NamedExpressionAccessor, error) {
+	if err := in.validateVariables(); err != nil {
+		return nil, err
+	}
+
+	vars := make([]cel.NamedExpressionAccessor, len(in.Variables))
+	for i, v := range in.Variables {
+		vars[i] = &validatingadmissionpolicy.Variable{
+			Name:       v.Name,
+			Expression: v.Expression,
+		}
+	}
+
+	return vars, nil
+}
+
+func (in *Source) GetV1Alpha1Variables() ([]admissionv1alpha1.Variable, error) {
+	var variables []admissionv1alpha1.Variable
```

Comment:
do we not need to validate variables here as well?

Reply:
added